Source/Login/Backend/configuration/viper/providers_configuration.go

```diff
@@ -0,0 +1,71 @@
+package viper
+
+import (
+	"net/url"
+
+	"dolittle.io/login/providers"
+	"github.com/spf13/viper"
+)
+
+const (
+	providersKey         = "providers"
+	providersNameKey     = "name"
+	providersImageURLKey = "image_url"
+)
+
+var (
+	defaultProviderName     = "UNKNOWN PROVIDER"
+	defaultProviderImageURL = ""
+)
+
+func newNilProvider() *providers.ProviderConfiguration {
+	return &providers.ProviderConfiguration{
+		Name:     defaultProviderName,
+		ImageURL: nil,
+	}
+}
+
+type providersConfiguration struct{}
+
+func (c *providersConfiguration) Providers() providers.Providers {
+	if !viper.IsSet(providersKey) {
+		return nil
+	}
+	return getProviderConfigurationMap(viper.GetStringMap(providersKey))
+}
+
+func getProviderConfigurationMap(in map[string]interface{}) providers.Providers {
+	providerConfigMap := make(map[providers.ProviderID]*providers.ProviderConfiguration, len(in))
+	for provider, config := range in {
+		providerConfigMap[provider] = getProviderConfiguration(provider, config)
+	}
+	return providers.Providers(providerConfigMap)
+}
+
+func getProviderConfiguration(providerID providers.ProviderID, config interface{}) *providers.ProviderConfiguration {
+	configMap, ok := config.(map[string]interface{})
+	if !ok {
+		return newNilProvider()
+	}
+	name, ok := configMap[providersNameKey].(string)
+
+	if !ok {
+		return newNilProvider()
+	}
+
+	imageURLString, ok := configMap[providersImageURLKey].(string)
+
+	if !ok {
+		return newNilProvider()
+	}
+
+	imageURL, err := url.Parse(imageURLString)
+	if err != nil {
+		return newNilProvider()
+	}
+
+	return &providers.ProviderConfiguration{
+		Name:     name,
+		ImageURL: imageURL,
+	}
```

Comment:
In stead of returning the default NilProvider thing, I suggest also returning an error or just a bool from this function in addition to the provider, and if it is not all ok => don't add it to the map.

Reply:
Fixed by https://github.com/dolittle-platform/Authentication/pull/10/commits/68d3ca5afdfe219c74055ab494857bcdec2339be